aten/src/ATen/core/type.cpp

```diff
@@ -287,6 +287,16 @@ c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2) {
   return c10::nullopt;
 }
 
+c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2, bool default_to_any) {
+  auto unified = unifyTypesImpl(t1, t2);
+
+  if (default_to_any && !unified) {
+    return c10::optional<TypePtr>(AnyType::get());
```

Comment:
nit: i dont think you actually need the cast here, you only needed the cast for pybind bc it was auto-inferring the return type